Lib/pdb.py

```diff
@@ -436,6 +438,20 @@ def _cmdloop(self):
             except KeyboardInterrupt:
                 self.message('--KeyboardInterrupt--')
 
+    def _validate_file_mtime(self):
+        """Check if the source file of the current frame has been modified since
+        the last time we saw it. If so, give a warning."""
+        try:
+            filename = self.curframe.f_code.co_filename
+            mtime = os.path.getmtime(filename)
+        except Exception:
+            return
+        if (filename in self._file_mtime_table and
+            mtime != self._file_mtime_table[filename]):
+            self.message(f"*** WARNING: file '{filename}' was edited after pdb started, "
```

Comment:
What do you think about this?  My concern is that it's confusing when you say the issue was the it was edited after pdb started, and then talk about restarting the program. It could be interpreted as restarting pdb, in the context of the first part.


```suggestion
            self.message(f"*** WARNING: file '{filename}' was edited, "
```

Reply:
At first "after pdb started" was to add a timestamp because "file was edited" was too dry and had very little information. However, with the message we added later, I think we can remove that part and still convey enough message to the user.